src/main/java/com/mic/betterslimes/entity/slimes/KingSlime.java

```diff
@@ -26,128 +32,389 @@
-	}
-
-	@Nullable
-	protected ResourceLocation getLootTable() {
-		return BetterSlimes.kingSlimeLT;
-	}
+                KnightSlime b;
+                for (int x = 0; x < 4; x++) {
+                    b = new KnightSlime(this.world);
+                    b.setSlimeSize(2, true);
+                    b.setLocationAndAngles(this.posX + rand.nextInt(10) - 5, this.posY + rand.nextInt(1) + 1,
+                            this.posZ + rand.nextInt(10) - 5, this.rotationYaw, this.rotationPitch);
+                    this.world.spawnEntity(b);
+                }
+            }
+
+            this.setSpawnTime(j1);
+        } else {
+            this.setSpawnTime(240);
+        }
+        super.updateAITasks();
+        this.bossInfo.setPercent(this.getHealth() / this.getMaxHealth());
+    }
+
+    @Override
+    public void addTrackingPlayer(EntityPlayerMP player) {
+        super.addTrackingPlayer(player);
+        this.bossInfo.addPlayer(player);
+
+    }
+
+    public void setCustomNameTag(String name) {
+        super.setCustomNameTag(name);
+        this.bossInfo.setName(this.getDisplayName());
+    }
+
+    public void writeEntityToNBT(NBTTagCompound compound) {
+        super.writeEntityToNBT(compound);
+        if (configLoaded) {
+            compound.setInteger("Spawn", this.getSpawnTime());
+            compound.setInteger("leapCooldown", leapCooldown);
+            compound.setInteger("leapWarning", leapWarning);
+            compound.setFloat("leapVelocityMultiplierY", leapVelocityMultiplierY);
+            compound.setFloat("leapVelocityMultiplierXZ", leapVelocityMultiplierXZ);
+            compound.setFloat("explodeDamage", explodeDamage);
+            compound.setInteger("explodeRange", explodeRange);
+            compound.setBoolean("spawnMinions", spawnMinions);
+            nbtSet = true;
+        }
+    }
+
+    public void readEntityFromNBT(NBTTagCompound compound) {
+        super.readEntityFromNBT(compound);
+        this.setSpawnTime(compound.getInteger("Spawn"));
+        if (this.hasCustomName()) {
+            this.bossInfo.setName(this.getDisplayName());
+        }
+        if (configLoaded  && nbtSet) {
+            leapCooldown = compound.getInteger("leapCooldown");
+            leapWarning = compound.getInteger("leapWarning");
+            leapVelocityMultiplierY = compound.getFloat("leapVelocityMultiplierY");
+            leapVelocityMultiplierXZ = compound.getFloat("leapVelocityMultiplierXZ");
+            explodeDamage = compound.getFloat("explodeDamage");
+            explodeRange = compound.getInteger("explodeRange");
+            spawnMinions = compound.getBoolean("spawnMinions");
+        }
+    }
+
+    public void removeTrackingPlayer(EntityPlayerMP player) {
+        super.removeTrackingPlayer(player);
+        this.bossInfo.removePlayer(player);
+    }
+
+    @Override
+    protected EntityBetterSlime createInstance() {
+        return new BlueSlime(this.world);
```

Comment:
Where's this used, and why doesn't it return an object with the KingSlime class? (Was it just from the fork? Might as well fix it here.)